MoppApp/MoppApp/CryptoContainerViewController.swift

```diff
@@ -0,0 +1,152 @@
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+    }
+    
+    func reloadCryptoData() {
+        if (container.addressees.count > 0 && state == .opened){
+            self.sections = ContainerViewController.sectionsEncrypted
+        } else if (container.addressees.count > 0) {
+            self.sections = ContainerViewController.sectionsWithAddresses
+        } else {
+            self.sections = ContainerViewController.sectionsNoAddresses
+        }
+        self.reloadData()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        containerViewDelegate = self
+        cryptoContainerViewDelegate = self
+        delegate = self
+        if (container == nil) {
+            self.sections = ContainerViewController.sectionsEncrypted
+        } else {
+            self.sections = ContainerViewController.sectionsNoAddresses
+        }
+        
+    }
+}
+
+extension CryptoContainerViewController : CryptoContainerViewControllerDelegate {
+    func startEncrypting() {
+        startEncryptingProcess()
+    }
+    
+    func removeSelectedAddressee(index: Int) {
+        container.addressees.removeObject(at: index)
+        reloadCryptoData()
+    }
+    
+    func getAddressee(index: Int) -> Any {
+        return container.addressees[index]
+    }
+    
+    func getAddresseeCount() -> Int {
+        return container.addressees.count
+    }
+    
+    func addAddressees() {
+        let addresseeController = UIStoryboard.container.instantiateViewController(of: AddresseeViewController.self)
+        addresseeController.addresseeViewControllerDelegate = self
+        addresseeController.selectedAddressees = container.addressees
+        landingViewController.importProgressViewController.dismissRecursively(animated: false, completion: {
+            self.navigationController?.pushViewController(addresseeController, animated: true)
+        })
+    }
+}
+
+extension CryptoContainerViewController : AddresseeViewControllerDelegate {
+    func addAddresseeToContainer(selectedAddressees: NSMutableArray) {
+        container.addressees = selectedAddressees
+        self.navigationController?.popViewController(animated: true)
+        reloadCryptoData()
+    }
+}
+
+extension CryptoContainerViewController : ContainerViewControllerDelegate {
+    
+    func removeDataFile(index: Int) {
+        confirmDeleteAlert(
+            message: L(.datafileRemoveConfirmMessage),
+            confirmCallback: { [weak self] (alertAction) in
+                self?.notifications = []
+                self?.updateState(.loading)
+                self?.updateState((self?.isCreated)! ? .created : .opened)
+                self?.container.dataFiles.removeObject(at:index)
+                self?.reloadData()
+        })
+
+    }
+    
+    func getDataFileOriginFilename(index: Int) -> String {
+        return ((container.dataFiles[index] as! CryptoDataFile).filePath as NSString).lastPathComponent
```

Comment:
Forced optional unwrapping is not recommended. Should use `guard let` or `if let` construct.